### ISSUE TITLE
Org "payment info" tab should have its own UI

### ISSUE THREAD

**Issue by seldo** [npm/newww#1587]
Clicking this payment info tab:

![screenshot 2015-11-30 11 01 08](https://cloud.githubusercontent.com/assets/185893/11481334/afb7d048-9751-11e5-9eef-36c21e2b03ec.png)

Currently takes you to https://www.npmjs.com/settings/billing. It should instead have UI that says "you are paying for X users, for Y per month" and the ability to cancel payment for that org. There should also be a link to https://www.npmjs.com/settings/billing to allow cancelation of everything, but that's not usually what they want.

**Comment by seldo** [comment]
Sending this one to @ncawthon for the look of the tab.

**Comment by ncawthon** [comment]
re B1 - why does the super-admin click "start paying" ?

**Comment by ncawthon** [comment]
in-air convo with @seldo hath resulted in the designations of "Full Access" and "Open Source" being the descriptors for the different tiers of members.

**Comment by ncawthon** [comment]
regarding the estimated bill - can we calculate this to know days of month / date added to not generalize this amount? I'm trying to recall a payment page where this is presented with such ambiguity. Also, if I take actions on this page, I'd like to see the amount of expected bill go up / down.

**Comment by seldo** [comment]
@ncawthon: it's not impossible to do this but I think we should do it later. At the moment the language should be something like "$X is your estimated monthly bill; this number can vary if you added or removed users within this billing cycle".

**Comment by ceejbot** [comment]
This is fairly important, so it'll get moved to "fix now" some time this week.

**Comment by seldo** [comment]
Nick's suggestion for better billing is going to turn into a new ticket, later today, so we don't lose track of it.

**Comment by rockbot** [comment]
@seldo, @ncawthon: the "Open Source" user state (as opposed to Full Access) is... odd. I understand that the notion is that an "Open Source" user has access to the org's public packages... but then again, so does any open source user (by the magic of said packages being public). I think it'd be better to call the users in this state "Inactive" - with a call to action to pay for them to make them active (or extend an invitation, but we can't quite do that yet).

Thoughts?

**Comment by seldo** [comment]
An "open source" user is different from somebody who isn't part of the org at all -- they can publish if they've been given that permission. So "inactive" isn't a good term, since they can install and maybe publish.

The goal I'm moving to with this language change is the idea of open-source organizations, where you pay for the ability to manage teams but, as long as they are only managing public packages, you can add as many un-paid users as you want.

"Inactive" and "unpaid" both sound like error states, but long-term an open-source user isn't an error state, it might be intended, so I wanted an intentional-sounding label for it. Make sense?

**Comment by seldo** [comment]
Even longer-term, we might make it possible to create an open-source organization that is free but cannot publish private packages at all, but that's a way off.

**Comment by jefflembeck** [comment]
Current things that need to be addressed:

There are three states to an org that we recognize right now:

- Active
- Canceled, but paid until end of billing cycle
- Canceled, and no longer paying

The "cancel org" button would affect #1 and it is easy enough to take care of.

The second one is waiting on https://github.com/npm/newww/pull/1615, which is passing partially on travis.

The third one requires some changes to our UI. Currently, we only show the first two types of org in our list of orgs, but if we want to handle restarting well, we're going to need to list all of them in our org list, and find a nice way to designate the difference. It is also relying upon #1615. 

Other than that, we have an issue with the process of revoking paying for a user, then re-extending and accepting the offer for paying for them, in that it is 3 api calls for each verified user in an org, and if any of them mess up along the way, we're in a bit of a jam. It would be best if we could handle this in one call, and one transaction in the license API.

**Comment by seldo** [comment]
This flowchart covers how this tab should behave right now. Later on we'll cover the "delete org" flow from here but it's not in scope right now:

![screenshot 2015-12-08 11 59 41](https://cloud.githubusercontent.com/assets/185893/11666828/4145754a-9da3-11e5-8fc2-865fadf9aa2d.png)

Obviously, for this to work it needs to be possible to get to this tab even if you are in @jefflembeck's third state. We have to modify the orgs list API call such that if you're super-admin of an org, your list of orgs contains all of those orgs, even if you are past the end of subscription for that org.

**Comment by seldo** [comment]
A wrinkle in the above: if your org is paid until Jan 15 and you re-start payment on Jan 1, the final box where we move your sponsorships over to the new license will create an error because your new license will *not have started yet*. This may require an extra parameter to the license creation to make it start *today* even though the billing date starts at the end of the previous period. This is probably most easily implemented as a "free trial" state, a concept already supported by stripe.

**Comment by jefflembeck** [comment]
API discussion: https://github.com/npm/license-api/issues/100

**Comment by ashleygwilliams** [comment]
hello @jefflembeck ! please let me know when this is complete for the purposes of documentation! thanks!
